### ISSUE TITLE
[New Resource]: DocumentDB - Elastic Clusters

### ISSUE THREAD

**Issue by vin-yu** [hashicorp/terraform-provider-aws#29514]
### Description

Add support for new type of Amazon DocumentDB Cluster - Elastic Clusters. 

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-docdbelastic-cluster.html

### Requested Resource(s) and/or Data Source(s)

aws_docdb_elastic

### References

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-docdbelastic-cluster.html 

https://docs.aws.amazon.com/documentdb/latest/developerguide/docdb-using-elastic-clusters.html 

### Would you like to implement a fix?

None

**Comment by emyller** [comment]
I don't understand how to use this. Does @amacf's PR enable a new resource type?

**Comment by tomwidmer** [comment]
> I don't understand how to use this. Does @amacf's PR enable a new resource type?

No, I think that requires further PRs, the initial PR just prepares the ground by adding the AWS client for the new service.

**Comment by c0st1nn** [comment]
any update on this ? looking forward to future PRs

**Comment by abhishekdas29** [comment]
> any update on this ? looking forward to future PRs

Likewise, this will be really helpful.

**Comment by narek** [comment]
Support for Elastic Clusters is added to Terraform AWSCC provider, @ewbankkit, @johnsonaj is there a way to have it in AWS provider as well? A lot our customers have requested to have an option of provisioning Elastic Clusters through AWS provider.

**Comment by johnsonaj** [comment]
> Support for Elastic Clusters is added to Terraform AWSCC provider, @ewbankkit, @johnsonaj is there a way to have it in AWS provider as well? A lot our customers have requested to have an option of provisioning Elastic Clusters through AWS provider.

@narek PR #31033 that closed this issue adds the `aws_docdbelastic_cluster` resource to the AWS provider. This will be released in the upcoming version.

**Comment by narek** [comment]
Cool, thank you @johnsonaj!
